### DISCUSSION TITLE
What are `Derrno` and `Errno` ?

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#653] [Q&A] [answered]
Could you provide more explanation about what `Derrno` and `Errno` are and what is the difference between the two - https://github.com/opendcs/opendcs/blob/master/src/main/java/lrgs/ldds/ServerError.java#L70?

**Comment by MikeNeilson** [comment]
I'll have to look more later but it appears all usages are just 0. So i suspect whatever meanimg was intended never came to fruition.

**Comment by ghost** [comment]
The `Derrno` are later set using [src/main/java/lrgs/common/LrgsErrorCode.java](https://github.com/opendcs/opendcs/blob/master/src/main/java/lrgs/common/LrgsErrorCode.java). I haven't figured out what `Errno` is though.

**Reply by MikeNeilson** [reply]
I suspect it's the same meaning. Looking at the ServerError.java again I saw this: https://github.com/opendcs/opendcs/blob/9eeba5bfa767862ae1866266ab36c37136177033/src/main/java/lrgs/ldds/ServerError.java#L143

pretty sure that second append should actually be with errno instead of Derrno. And the function above looks like it parses the msg to pull numbers out of there.

Unfortunately you've found a section that's about as clear as mud at the moment. But it looks like it's a way for the server to show like combined error messages.

**Comment by ghost** [comment, marked as answer]
Looks like they are server code and system code - https://github.com/opendcs/dcs_standards/blob/main/source/dds-14.rst#normal-and-error-responses.

**Reply by MikeNeilson** [reply]
That makes sense to me. Pretty sure that section of code I reference is just wrong. I've opened #664 to track fixing it.
